packages/firestore/src/api/user_data_reader.ts

```diff
@@ -138,11 +138,14 @@ interface ContextSettings {
    * A path within the object being parsed. This could be an empty path (in
    * which case the context represents the root of the data being parsed), or a
    * nonempty path (indicating the context represents a nested location within
-   * the data).
+   * the data). Defaults to null.
    */
-  readonly path: FieldPath | null;
-  /** Whether or not this context corresponds to an element of an array. */
-  readonly arrayElement: boolean;
+  readonly path?: FieldPath;
+  /**
+   * Whether or not this context corresponds to an element of an array.
+   * Defaults to false.
```

Comment:
Consider re-wording this as something like "if undefined, it will be treated as false" because the way that it is worded suggests to me that it gets assigned a default value of false somehow, which I do not think is accurate.

Reply:
Changed to `If not set, elements are treated as if they were outside of arrays.`

Reply:
Love it.